stream.go

```diff
@@ -107,13 +75,35 @@ type ClientStream interface {
 	// CloseSend closes the send direction of the stream. It closes the stream
 	// when non-nil error is met.
 	CloseSend() error
-	// Stream.SendMsg() may return a non-nil error when something wrong happens sending
-	// the request. The returned error indicates the status of this sending, not the final
-	// status of the RPC.
+	// Context returns the context for this stream.
+	Context() context.Context
+	// SendMsg is generally called by generated code. On error, SendMsg aborts
+	// the stream and returns an io.EOF; the status may be determined by
+	// inspecting the error returned by calling RecvMsg.
+	//
+	// SendMsg blocks until:
+	//   - It schedules m with the transport, or
+	//   - The stream is done, or
+	//   - The stream breaks.
+	//
+	// SendMsg does not wait for an ack. An untimely stream closure can result
+	// in any buffered messages along the way (including those in the
+	// client-side buffer that comes with gRPC by default) being lost. To
+	// ensure delivery, users must call RecvMsg until receiving an EOF before
+	// cancelling the context.
 	//
-	// Always call Stream.RecvMsg() to drain the stream and get the final
-	// status, otherwise there could be leaked resources.
-	Stream
+	// It's safe to have a goroutine calling SendMsg and another goroutine
+	// calling RecvMsg on the same stream at the same time. It is not safe
+	// to call SendMsg on the same stream in different goroutines.
+	SendMsg(m interface{}) error
+	// RecvMsg blocks until it receives a message or the stream is done. On
```

Comment:
Let's remove the client/server language:

RecvMsg blocks until it receives a message into m or the stream is done.  It returns io.EOF when the stream completes successfully. On any other error, it the stream is aborted and the error contains the RPC status.

It is safe to have a goroutine calling SendMsg and another goroutine calling RecvMsg on the same stream at the same time, but it is not safe to call RecvMsg on the same stream in different goroutines.

Reply:
Done.